rclpy/rclpy/parameter.py

```diff
@@ -0,0 +1,134 @@
+                return isinstance(parameter_value, bool)
+            if Parameter.Type.INTEGER == self:
+                return isinstance(parameter_value, int)
+            if Parameter.Type.DOUBLE == self:
+                return isinstance(parameter_value, float)
+            if Parameter.Type.STRING == self:
+                return isinstance(parameter_value, str)
+            if Parameter.Type.BYTE_ARRAY == self:
+                return isinstance(parameter_value, list) and \
+                    all(isinstance(v, bytes) and len(v) == 1 for v in parameter_value)
+            if Parameter.Type.BOOL_ARRAY == self:
+                return isinstance(parameter_value, list) and \
+                    all(isinstance(v, bool) for v in parameter_value)
+            if Parameter.Type.INTEGER_ARRAY == self:
+                return isinstance(parameter_value, list) and \
+                    all(isinstance(v, int) for v in parameter_value)
+            if Parameter.Type.DOUBLE_ARRAY == self:
+                return isinstance(parameter_value, list) and \
+                    all(isinstance(v, float) for v in parameter_value)
+            if Parameter.Type.STRING_ARRAY == self:
+                return isinstance(parameter_value, list) and \
+                    all(isinstance(v, str) for v in parameter_value)
+            return False
+
+    @classmethod
+    def from_rcl_interface_parameter(cls, rcl_param):
+        value = None
+        type_ = Parameter.Type(value=rcl_param.value.type)
+        if Parameter.Type.BOOL == type_:
+            value = rcl_param.value.bool_value
+        elif Parameter.Type.INTEGER == type_:
+            value = rcl_param.value.integer_value
+        elif Parameter.Type.DOUBLE == type_:
+            value = rcl_param.value.double_value
+        elif Parameter.Type.STRING == type_:
+            value = rcl_param.value.string_value
+        elif Parameter.Type.BYTE_ARRAY == type_:
+            value = rcl_param.value.byte_array_value
+        elif Parameter.Type.BOOL_ARRAY == type_:
+            value = rcl_param.value.bool_array_value
+        elif Parameter.Type.INTEGER_ARRAY == type_:
+            value = rcl_param.value.integer_array_value
+        elif Parameter.Type.DOUBLE_ARRAY == type_:
+            value = rcl_param.value.double_array_value
+        elif Parameter.Type.STRING_ARRAY == type_:
+            value = rcl_param.value.string_array_value
+        return cls(rcl_param.name, type_, value)
+
+    def __init__(self, name, type_, value=None):
+        if not isinstance(type_, Parameter.Type):
+            raise TypeError("type must be an instance of '{}'".format(repr(Parameter.Type)))
+
+        if not type_.check(value):
+            raise ValueError("Type '{}' and value '{}' do not agree".format(type_, value))
+
+        self._type_ = type_
+        self._name = name
+        self._value = value
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def type_(self):
```

Comment:
@mikaelarguedas @sloretz the property is now accessed as `type_`. I had initially let this slide since the generated interface code uses `type` in python currently and I know that surfaced in discussion recently but can't find it to figure out where we came down.

Reply:
Makes sense. And the change is in so all's well.